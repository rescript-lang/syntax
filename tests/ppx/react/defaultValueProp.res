module C0 = {
  let a = 1
  @react.component
  let make = (~a=2, ~b=a*2) => React.int(a + b)
}

module C1 = {
  @react.component
  let make = (~a=2, ~b) => React.int(a + b)
}
