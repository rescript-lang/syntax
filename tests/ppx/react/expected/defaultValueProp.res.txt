module C0 = {
  let a = 1
  type props<'a, 'b> = {a?: 'a, b?: 'b}

  @react.component
  let make = (props: props<'a, 'b>) => {
    let b = switch props.b {
    | Some(b) => b
    | None => a * 2
    }
    and a = switch props.a {
    | Some(a) => a
    | None => 2
    }

    React.int(a + b)
  }
  let make = {
    let \"DefaultValueProp$C0" = (props: props<_>) => make(props)

    \"DefaultValueProp$C0"
  }
}

module C1 = {
  type props<'a, 'b> = {a?: 'a, b: 'b}

  @react.component
  let make = (props: props<'a, 'b>) => {
    let b = props.b
    and a = switch props.a {
    | Some(a) => a
    | None => 2
    }

    React.int(a + b)
  }
  let make = {
    let \"DefaultValueProp$C1" = (props: props<_>) => make(props)

    \"DefaultValueProp$C1"
  }
}
